src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -943,11 +943,32 @@ internal Type FastBindToType(string assemblyName, string typeName)
             if (entry == null || entry.AssemblyName != assemblyName)
             {
                 Assembly assm = null;
-                try
+                if (_isSimpleAssembly)
                 {
-                    assm = Assembly.Load(new AssemblyName(assemblyName));
+                    try
+                    {
+                        assm = Assembly.Load(assemblyName);
+                    }
+                    catch { }
+
+                    if (assm == null)
```

Comment:
is already checking here: https://github.com/dotnet/corefx/pull/20552/files#diff-b4b2fd9840862f03eae907efcd988a68R1011